enhancements/testing/ephemeral-agent-installer-e2e-tests.md

```diff
@@ -0,0 +1,337 @@
+logic will land in dev-scripts alongside the pre environment setup steps.
+
+#### Tests
+
+For the actual test scenarios themselves we should use Ginkgo/Gomega. This is 
+similar to the rest of Openshift. Well most of (if not all) the product is written in Golang
+so it makes sense to use a Golang framework for these.
+
+There is a repository created as a preliminary place to put these. Plan to use a similar
+structure to openshift-tests in this repository.
+
+We will be including existing e2e tests that have been created to validate the OCP cluster
+installed by the new agent based installer.
+
+#### Teardown
+
+Currently there is a `make clean` flow for starting new locally. We will need to add similar
+functionally for the flows that will be used for the agent based installer
+
+### Identified Overlap Points for Agent Installer
+
+During the discussions with the Metal Platform team it was identified that the following
+components of dev-scripts are the likely points of overlap for the new agent installer project:
+
+* The [01_install_requirements.sh](https://github.com/openshift-metal3/dev-scripts/blob/master/01_install_requirements.sh) and [02_configure_host.sh](https://github.com/openshift-metal3/dev-scripts/blob/master/02_configure_host.sh) scripts that are run using `make`
+* Ansible playbooks to set up VMs such as: [vm_setup_vars.yml](https://github.com/openshift-metal3/dev-scripts/blob/master/vm_setup_vars.yml)
+
+These will be the things that we will focus on first when refactoring parts of dev-scripts.
+
+### Open Questions [optional]
+
+### Test Plan
+
+* e2e tests for the three install scenarios of cluster 0 (SNO, Compact, HA)
+* Run Openshift compliance tests against these installed clusters
+* Tests in dev-scripts to cover these changes made for agent-installer
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+* Ability to utilize the e2e framework locality to aid in development
+* Openshift compliance tests are run against the installed clusters
+* The e2e framework is on boarded into Openshift CI for the team's use
+
+#### Tech Preview -> GA
+
+* Add upgrade/downgrade e2e tests
+
+#### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+### Version Skew Strategy
+
+* Currently the dev-scripts repository is not versioned. This will stay the same.
+
+* However for the e2e test scenarios themselves it makes sense to version these.
+Since the ephemeral agent based installer is targeted to be released with OCP it would 
+be easiest to version the e2e framework the same as OCP x.y. This is the strategy
+that openshift-tests takes.
+
+### Operational Aspects of API Extensions
+
+#### Failure Modes
+
+#### Support Procedures
+
+## Implementation History
+
+## Drawbacks
+
+The biggest argument against it is that by creating a new framework we don't
+inherit the work that was done previously in validating Assisted Installer.
```

Comment:
Is this talking about creating a new test suite, or reusing dev-scripts instead of the AI test infra repo?

Reply:
It's talking about creating a new test suite.